src/react-apps/applications/ux-editor/src/components/config/EditModalContent.tsx

```diff
@@ -145,14 +131,51 @@ class EditModalContentComponent extends React.Component<IEditModalContentProps,
     return element.Texts.Label;
   }
 
-  public getTextResourceKeys = (): any[] => {
-    if (!this.props.textResources) {
-      return [];
-    }
+  public renderSelectDataModelBinding(): JSX.Element {
+    return (
+      <div>
+        <Typography gutterBottom={false} className={this.props.classes.inputHelper}>
+          {this.props.language.ux_editor.modal_properties_data_model_helper}
+        </Typography>
+        <SelectDataModelComponent
+          selectedElement={this.props.component.dataModelBinding}
+          onDataModelChange={this.handleDataModelChange}
+          language={this.props.language}
+        />
+      </div>
+    );
+  }
 
-    return (this.props.textResources.map((resource) => {
-      return resource.id;
-    }));
+  public renderSelectTextFromResources = (
+    labelText: string,
+    onChangeFunction: (e: any) => void,
+    placeholder?: string,
+    truncateLimit: number = 80,
+  ): JSX.Element => {
+    const textRecources: any = [];
+    this.props.textResources.map((resource, index) => {
+      const option = truncate(resource.value, truncateLimit);
+      textRecources.push({ value: resource.id, label: option.concat('\n(', resource.id, ')') });
+    });
+    return (
+      <div>
+        <Typography gutterBottom={false} className={this.props.classes.inputHelper}>
+          {this.props.language.ux_editor[labelText]}
```

Comment:
Right, the fallback language is loaded when the api call to get the languages fail. But maybe we should use the new method for extracting the text, so that the key is shown if it's missing instead of just nothing.

Reply:
Should change this when the method is available then.